.github/update_labels.py

```diff
@@ -0,0 +1,135 @@
+    for key in entry_dict:
+        if key not in actual_label or actual_label[key] == "":
+            actual_label[key] = entry_dict[key]
+
+    return actual_label
+
+# Given a list of labels from .github/labels, return a mapping from the label name to the label metadata.
+def map_name_to_label(labels_from_file: list) -> dict:
+    name_to_label = {}
+    for label in labels_from_file:
+        name_to_label[label["name"]] = label
+    return name_to_label
+
+# Return YAML context with output parameters set.
+def get_yaml_context() -> YAML:
+    yaml_context = YAML()
+    yaml_context.default_flow_style = False
+    yaml_context.indent(mapping=2, sequence=4, offset=2)
+    yaml_context.sort_keys = False
+    yaml_context.line_break = "\n"
+    return yaml_context
+
+# Given a file name, open associated input and output files.
+def get_yaml_input_output(file_name: str, yaml_context: YAML):
+    base_label_dir = "labels"
+    label_path = f"{base_label_dir}/{file_name}.yaml"
+    template_path = "LABEL_TEMPLATE.yaml"
+    label_file_exists = os.path.isfile(label_path)
+    old_yaml_path = label_path if label_file_exists else template_path
+
+    with open(old_yaml_path, "r+") as old_yaml_file: # open exisiting YAML file or template.
+        labels_from_file = yaml_context.load(old_yaml_file)
+
+    output_file = open(label_path, "w")
+    
+    return labels_from_file, output_file, yaml_context
+
+# Updates a specific file in .github/labels by either reading the existing file from disk or creating a new label.
+# In either case, an in-memory object with the "old" version of the data is created and subsequently updated before being written back.
+def update_repo_for_file(file_name: str, actual_labels: list) -> None:
+    yaml_context = get_yaml_context() # used for loading / dumping yaml files
+    labels_from_file, output_file, yaml_dumper_context = get_yaml_input_output(file_name, yaml_context)
+
+    # actual_labels may not contain all entries found in the file from .github/labels.
+    # If it contains a specific field (such as description), that field may be empty.
+    # Here, we make sure that empty or missing fields do not overwrite feels that are present.
+    labels_from_file_dict = map_name_to_label(labels_from_file["labels"])
+    for label in actual_labels:
+        # Ensure we don't overwrite existing data with blank data.
+        label = update_empty_entries(labels_from_file_dict, label)
+
+    # Write new labels to file.
+    labels_from_file["labels"] = actual_labels
+    yaml_dumper_context.dump(labels_from_file, output_file)
+    output_file.close()
+
+
+# Updates .github/labels with labels that are retrieved via the Github API.
+def update_repo(labels: defaultdict(list)):
+    for file_name in labels:
+        update_repo_for_file(file_name, labels[file_name])
+
+# Usage: python update_labels.py
+if __name__ == "__main__":
+    labels = get_labels()
+    update_repo(labels)
```

Comment:
```suggestion
    update_repo(labels)

```

Missing newline at the end